src/Microsoft.IdentityModel.Tokens/SecurityTokenDescriptor.cs

```diff
@@ -94,6 +94,16 @@ public class SecurityTokenDescriptor
         /// </summary>
         public IDictionary<string, object> AdditionalInnerHeaderClaims { get; set; }
 
+        /// <summary>
+        /// Gets or sets the <see cref="HashSet{T}"/> which contains the default header claims that should be excluded from the JWT header.
+        /// </summary>
+        /// <remarks>
+        /// The default header claims are 'alg', 'kid', 'x5t', 'enc', and 'zip'.
+        /// </remarks>
+#pragma warning disable CA2227 // Collection properties should be read only
+        public ISet<string> ExcludedDefaultHeaderClaims { get; set; }
```

Comment:
What is the relationship between if kid is included here and "IncludeKeyIdInHeader"?

Can that be included in the doc string here and likely in "IncludeKeyIdInHeader"?

Reply:
Ideally we'd remove `IncludeKeyIdInHeader` and let this supersede it.

Reply:
Let's at least update the doc string for now.